src/test/java/build/buildfarm/ByteStreamServiceTest.java

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm;
+
+import build.buildfarm.common.Digests;
+import build.buildfarm.server.BuildFarmInstances;
+import build.buildfarm.server.ByteStreamService;
+import com.google.bytestream.ByteStreamProto.WriteRequest;
+import com.google.bytestream.ByteStreamProto.WriteResponse;
+import com.google.devtools.remoteexecution.v1test.Digest;
+import com.google.protobuf.ByteString;
+import io.grpc.stub.StreamObserver;
+import java.util.Random;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ByteStreamServiceTest {
+
+  private static byte[] generateFileData(Random random) {
+    byte[] data = new byte[1024 * 1024];
+    random.nextBytes(data);
+    return data;
+  }
+
+  @Test
+  public void testValidWrite() {
+    BuildFarmInstances instances = mock(BuildFarmInstances.class);
+
+    ByteStreamService service = new ByteStreamService(instances);
+
+    StreamObserver<WriteResponse> responseObserver = mock(StreamObserver.class);
+    verify(responseObserver, never()).onError(any());
+
+    StreamObserver<WriteRequest> requestObserver = service.write(responseObserver);
+
+    String instance = "foo_instance";
+    String uuid = "1d4e65cd-7c0d-44ca-b49b-b912988c8967"; // dummy value
+
+    Random random = new Random(49);
+    byte[] data = generateFileData(random);
+    ByteString bytes = ByteString.copyFrom(data);
+    Digest digest = Digests.computeDigest(bytes);
+    String hash = digest.toString();
+    int size = data.length;
+
+    WriteRequest request = WriteRequest.newBuilder()
+      .setResourceName(String.format("%s/uploads/%s/blobs/%s/%d", instance, uuid, hash, size))
```

Comment:
Google style is four spaces indentation on continuation lines

Reply:
Ok, but then we should really add a pre-commit hook to the repo. I guess you have one floating around at Google?

Reply:
@abergmeier-dsfishlabs, I highly recommend using https://github.com/google/google-java-format which automatically formats the code using Google style. Ideally this should indeed be configured as part of presubmit hook or some other automated way.

Reply:
@ttsugriy I believe we can count Andreas out of maintaining this.  If you'd like to pick up the mantle and carry it forward, I will review in turn. Otherwise I will close this when I get finished with larger scale code coverage-based test additions.